### PR TITLE
adds a bookmark warning to the top of the app

### PR DESCRIPTION
**Pull request by robbestad**
Added a bookmark warning div that displays a bookmark warning as shown in the Figma design.
bilde

The feature is implemented by pushing the content down by 2 rem units. When displayed, the total page height is shrinked by the same amount to avoid showing a scrollbar. Tested in Firefox and Chrome.

### CODE REVIEW

**Comment by piersss** [comment]
Will check later today!

File: src/components/BookmarkWarning/BookmarkWarning.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import Icon from "../Icon/Icon";
+import { ButtonX, StyledBookmarkWarning, Text } from "./BookmarkWarning.styles";
+
+export type BookmarkWarningProps = {
+  /**
+   * if hidden, return null so that nothing gets rendered
+   */
+  hidden?: boolean;
+} & React.ButtonHTMLAttributes<HTMLButtonElement>;
```
Comment (piersss) [inline src/components/BookmarkWarning/BookmarkWarning.tsx:11]: BookmarkWarning should not be a button, so it shouldn't inherit HTMLButtonElement props.

**Review by piersss — CHANGES_REQUESTED** [review]
Looking good. Requested some minor changes.

File: src/components/BookmarkWarning/BookmarkWarning.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+export const BookmarkWarning = ({
+  hidden = false,
+  onClick,
+}: BookmarkWarningProps) => {
+  if (hidden) return null;
+  return (
+    <StyledBookmarkWarning>
+      <Text>
+        Make sure the URL is <strong>airswap.io</strong> - Press (Ctrl+D or
+        Cmd+D) to bookmark it to be safe.
```
[the remark this answers is not in the document]
Comment (robbestad) [reply src/components/BookmarkWarning/BookmarkWarning.tsx:22]: Sure! Which one the existing locale files are appropriate do you think, if any?

File: src/components/BookmarkWarning/BookmarkWarning.styles.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+export const ButtonX = styled.button`
+  position: absolute;
+  right: 2rem;
+  color: ${(props) => props.theme.colors.alwaysWhite};
+  pointer-events: ${(props) => (props.disabled ? "none" : "visible")};
+  cursor: ${(props) => (props.disabled ? "none" : "pointer")};
+`;
+
+export const StyledBookmarkWarning = styled.button<BookmarkWarningProps>`
```
[the remark this answers is not in the document]
Comment (robbestad) [reply src/components/BookmarkWarning/BookmarkWarning.styles.tsx:17]: ok fixed

File: src/components/BookmarkWarning/BookmarkWarning.styles.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  font-size: 1rem;
+  font-weight: 600;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  color: ${(props) => props.theme.colors.alwaysWhite};
+  background: ${(props) => props.theme.colors.primary};
+  pointer-events: ${(props) => (props.disabled ? "none" : "visible")};
+  cursor: ${(props) => (props.disabled ? "none" : "pointer")};
+  z-index: 1;
```
[the remark this answers is not in the document]
Comment (robbestad) [reply src/components/BookmarkWarning/BookmarkWarning.styles.tsx:33]: Ok! Best to hide it I think. The text and the design might change a bit to look right on mobile.

File: src/components/BookmarkWarning/BookmarkWarning.tsx (hunk 1 above)
Comment (robbestad) [reply src/components/BookmarkWarning/BookmarkWarning.tsx:11]: Ok, fixed

File: src/components/BookmarkWarning/BookmarkWarning.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (piersss) [reply src/components/BookmarkWarning/BookmarkWarning.tsx:22]: wallet.json should be fine

File: src/components/BookmarkWarning/BookmarkWarning.styles.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (piersss) [reply src/components/BookmarkWarning/BookmarkWarning.styles.tsx:32]: Add text-overflow ellipsis to `Text`, so it's still responsive regardless of content length.

```
text-overflow: ellipsis;
text-align: center;
overflow: hidden;
```

And padding to `StyledBookmarkWarning`:

```
padding-left: 1rem;
padding-right: 3.5rem;
```

[the remark this answers is not in the document]
Comment (robbestad) [reply src/components/BookmarkWarning/BookmarkWarning.styles.tsx:32]: Done!

[the remark this answers is not in the document]
Comment (piersss) [reply src/components/BookmarkWarning/BookmarkWarning.styles.tsx:32]: Nice. I don't see any new commits though. :)

**Review by piersss — APPROVED** [review]

**Review by gpxl-dev — CHANGES_REQUESTED** [review]
The Figma design has 8px margin top and bottom around the text which makes the blue bar 40px high in total. At the moment the text looks a bit cramped in this PR.

**Review by gpxl-dev — APPROVED** [review]
LGTM!

**Pull request merged**